02-typescript/02-todo-list/src/main.ts

```diff
@@ -28,33 +31,62 @@ const toDos: State = [
   },
 ];
 
+
 const crtToDo = () => {
   getDiv.innerHTML= "";
-  let crtInput = document.createElement("input");
+ 
   crtInput.setAttribute("type","text");
   crtInput.innerHTML= " ";
   getDiv.appendChild(crtInput);
-  let crtUl = document.createElement("ul");
-  getDiv.appendChild(crtUl)
+  
+  getDiv.appendChild(crtUl);
+  
+  let crtButton = document.createElement('button');
+  listTodo();
+  crtButton.innerText ="Add";
+  getDiv.appendChild(crtButton)
+  crtButton.addEventListener("click",addTodo);
+  crtFilter?.addEventListener("click",function(){
+    listTodo(crtFilter?.checked);
```

Comment:
You do not need the `?` in these two lines, because `crtFilter` is already guaranteed not to be `null` at this point.

Reply:
(The `?` in this case is used to ignore a potentially `null` value.)